// This defines the lemmatization protocol buffers for NLP in ATLAS.ti.
syntax = "proto3";

// service LanguageDetection {
// 	rpc Language(Document) returns(LanguageIdentifier) {}
// 	rpc LanguageSupported(Document) returns(Boolean) {}
// 	rpc LanguageSupported(LanguageIdentifier) returns(Boolean) {}
// }

service Lemmatizer {
	// The request will be split into words and lemmatized. Returns location of sentences that conatain ALL lemmatized words from the request.
	rpc LemmaSentencesForUnionisedRequest(Query) returns(Locations) {}
	// Return a sequence of all found lemmas and their frequencies in a given document. 
	rpc LemmasForDocument(Document) returns(LemmasAndFrequencies) {}
	rpc Test(TestString) returns(TestString) {}
}

message TestString {
	string message = 1;
}

message Document {
	string content = 1;
}

message Query {
	string searchTerm = 1;
	Document document = 2;
}

message Locations {
	message Location {
		int64 start = 1;
		int64 end = 2;
	}
	repeated Location locations = 1;
}

message LemmasAndFrequencies {
	repeated string lemma = 1;
	repeated int64 frequency = 2;
}